spring-pulsar/src/main/java/org/springframework/pulsar/config/AbstractPulsarListenerContainerFactory.java

```diff
@@ -118,7 +118,7 @@ public void afterPropertiesSet() {
 	@Override
 	public C createListenerContainer(PulsarListenerEndpoint endpoint) {
 		C instance = createContainerInstance(endpoint);
-		JavaUtils.INSTANCE.acceptIfNotNull(endpoint.getSubscriptionName(), instance::setBeanName);
+		JavaUtils.INSTANCE.acceptIfNotNull(endpoint.getId(), instance::setBeanName);
```

Comment:
@sobychacko let's chat about this change. I am not sure if you were using the sub name as the id always has the child container index on it or some other reason.

Reply:
We discussed this internally with @onobc. We decided to use `id` as the unique way to discover listeners. So this change is acceptable.